### ISSUE TITLE
HTML causes fenced code block detection issue

### ISSUE THREAD

**Issue by treyhunner** [rvben/rumdl#27]
When testing `file8.md` file:

    <div>
    <ul>
        <li>Variables are pointers in Python</li>
        <li>Assignment versus Mutation in Python</li>
    </ul>
    </div>

    ```pycon
    >>> numbers = [2, 1, 3, 4, 7]
    >>> numbers2 = [11, 18, 29]
    >>> name = "Trey"
    ```

Linting errors appear (incorrectly) about using fenced code blocks:

```
$ uv tool run rumdl check file8.md
topics/pointers.md:8:1: [MD046] Use fenced code blocks [*]
topics/pointers.md:9:1: [MD046] Use fenced code blocks [*]
topics/pointers.md:10:1: [MD046] Use fenced code blocks [*]
topics/pointers.md:11:1: [MD046] Use fenced code blocks [*]
topics/pointers.md:12:1: [MD046] Use fenced code blocks [*]

Issues: Found 5 issues in 1 file (30ms)
Run with `--fix` to automatically fix 5 of the 5 issues
```

Oddly, removing the second `<li>` line fixes the issue.

**Comment by rvben** [comment]
Hi @treyhunner,

Thanks for reporting this issue! The fix will be included in the next release. 

The problem was that the document structure analysis was detecting indented code blocks before HTML blocks. When your HTML had multiple indented lines, they were incorrectly identified as indented code blocks.

This has been fixed in commit 80aa2b1. The fix ensures that:
- HTML blocks are detected before computing code blocks
- Lines inside HTML blocks are excluded from indented code block detection

Thanks for the clear reproduction case!
